test/e2e/operator_test.go

```diff
@@ -338,6 +387,299 @@ func TestAWSLoadBalancerControllerWithCustomIngressClass(t *testing.T) {
+		Name:          aws.String("echoserver-acl"),
+		Scope:         wafv2types.ScopeRegional,
+		VisibilityConfig: &wafv2types.VisibilityConfig{
+			CloudWatchMetricsEnabled: false,
+			MetricName:               aws.String("echoserver"),
+			SampledRequestsEnabled:   false,
+		},
+	})
+	if err != nil {
+		t.Fatalf("failed to create aws wafv2 acl due to %v", err)
+	}
+	defer func() {
+		_, err = wafClient.DeleteWebACL(context.TODO(), &wafv2.DeleteWebACLInput{
+			Id:        aws.String(*acl.Summary.Id),
+			Name:      aws.String("echoserver-acl"),
+			LockToken: acl.Summary.LockToken,
+			Scope:     wafv2types.ScopeRegional,
+		})
+		if err != nil {
+			t.Fatalf("failed to delete aws wafv2 acl due to %v", err)
+		}
+	}()
+
+	t.Log("Creating Ingress Resource with default ingress class")
+	ingName := types.NamespacedName{Name: "echoserver", Namespace: testWorkloadNamespace}
+	ingAnnotations := map[string]string{
+		"alb.ingress.kubernetes.io/scheme":        "internet-facing",
+		"alb.ingress.kubernetes.io/target-type":   "instance",
+		"alb.ingress.kubernetes.io/wafv2-acl-arn": *acl.Summary.ARN,
+	}
+	echoIng := buildEchoIngress(ingName, "alb", ingAnnotations, echosvc)
+	err = retry.OnError(defaultRetryPolicy,
+		func(err error) bool {
+			t.Logf("retrying creation of echo ingress due to %v", err)
+			return !errors.IsAlreadyExists(err)
+		},
+		func() error { return kubeClient.Create(context.TODO(), echoIng) })
+	if err != nil && !errors.IsAlreadyExists(err) {
+		t.Fatalf("failed to ensure echo ingress %s: %v", echoIng.Name, err)
+	}
+	defer func() {
+		waitForDeletion(t, kubeClient, echoIng, defaultTimeout)
+	}()
+
+	var address string
+	if address, err = getIngress(t, kubeClient, defaultTimeout, ingName); err != nil {
+		t.Fatalf("did not get expected available condition for ingress: %v", err)
+	}
+
+	t.Logf("Testing aws load balancer for ingress traffic at address %s", address)
+	for i, rule := range echoIng.Spec.Rules {
+		clientPod := buildCurlPod(fmt.Sprintf("clientpod-%d", i), testWorkloadNamespace, rule.Host, address)
+		if err := kubeClient.Create(context.TODO(), clientPod); err != nil {
+			t.Fatalf("failed to create pod %s/%s: %v", clientPod.Namespace, clientPod.Name, err)
+		}
+
+		err = wait.PollImmediate(1*time.Second, 10*time.Minute, func() (bool, error) {
+			readCloser, err := kubeClientSet.CoreV1().Pods(clientPod.Namespace).GetLogs(clientPod.Name, &corev1.PodLogOptions{
+				Container: "curl",
+				Follow:    false,
+			}).Stream(context.TODO())
+			if err != nil {
+				t.Logf("failed to read output from pod %s: %v (retrying)", clientPod.Name, err)
+				return false, nil
+			}
+			scanner := bufio.NewScanner(readCloser)
+			defer func() {
+				if err := readCloser.Close(); err != nil {
+					t.Fatalf("failed to close reader for pod %s: %v", clientPod.Name, err)
+				}
+			}()
+			for scanner.Scan() {
+				line := scanner.Text()
+				if strings.Contains(line, "403 Forbidden") {
```

Comment:
I'm not sure that the presence of a 403 Forbidden is a good indicator that things are working as expected. If you really want to verify that the WAF is the reason for this, then the WAF rules should permit requests with only a certain header values and then you need to check with requests with and without those header values. Also the curl pod is just really not required. It's just adding bulk to the tests with no added benefit. WDYT?

Reply:
> I'm not sure that the presence of a 403 Forbidden is a good indicator that things are working as expected.

Regarding the 403 status check, WAFv2 provides a way to return a custom response when blocked, what would you say about using that? 

> Also the curl pod is just really not required. It's just adding bulk to the tests with no added benefit. WDYT?

The main benefit I see is with the amount of retries and the duration. Currently, it takes about 2-3min for the DNS to get registered and working properly. We can replicate the same with just HTTP clients, just with a few go routines (to handle multiple rules in the future). Also, we are missing an internal load balancer, which I was going to add as well.

Reply:
I'm open to retaining it only for the internal load balancer and replacing the rest. Although, would be better in a different PR.